client/client.go

```diff
@@ -80,4 +80,12 @@ type OpAMPClient interface {
 	// May be called anytime after Start(), including from OnMessage handler.
 	// nil values are not allowed and will return an error.
 	SetPackageStatuses(statuses *protobufs.PackageStatuses) error
+
+	// SetCustomMessage sets the custom message that will be sent to the Server. May be
+	// called anytime after Start(), including from OnMessage handler.
+	//
+	// If the CustomMessage is nil or it specifies a capability that is not listed in the
+	// CustomCapabilities provided in the StartSettings for the client, it will return an
+	// error.
+	SetCustomMessage(message *protobufs.CustomMessage) error
```

Comment:
Does this have the semantics as the other "Set" functions? Does it overwrite a previously set value if it was not already sent? I am not sure this will result in an ergonomic API. If you want to send multiple messages how would you know that the previous one is already sent?

Reply:
Good feedback. No, it schedules a send like the other APIs, but it does not guarantee that one has not already been added to the next message. I suppose we could add a queue that we drain or use a channel. Did you have something in mind?

Reply:
Thanks, I'll revisit this function.